redash/models.py

```diff
@@ -685,6 +688,9 @@ def should_schedule_next(previous_iteration, now, schedule):
             previous_iteration = normalized_previous_iteration - datetime.timedelta(days=1)
 
         next_iteration = (previous_iteration + datetime.timedelta(days=1)).replace(hour=hour, minute=minute)
+    else:
```

Comment:
Don't you prefer to have only cron-style syntax in the DB and use only croniter?

Also this needs more tests to make sure it acts as intended.